### PR TITLE
tap-to-retry failed thumbnail downloads

### PR DESCRIPTION
**Pull request by michaelkirk-signal**
PTAL @charlesmchen 

screen shot 2018-04-16 at 3 52 22 pm

### CODE REVIEW

File: Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.h [hunk 1]
```diff
@@ -7,10 +7,20 @@ NS_ASSUME_NONNULL_BEGIN
 @class DisplayableText;
 @class OWSBubbleStrokeView;
 @class OWSQuotedReplyModel;
+@class TSAttachmentPointer;
+@class TSQuotedMessage;
+
+@protocol OWSQuotedMessageViewDelegate
+
+- (void)didTapQotedReply:(OWSQuotedReplyModel *)quotedReply
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.h]: missing u in quoted.

File: Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m [hunk 2]
```diff
@@ -140,6 +140,25 @@ - (void)createContents
...
+            contentImageView.contentMode = UIViewContentModeScaleAspectFit;
+            contentImageView.userInteractionEnabled = YES;
+            contentImageView.tintColor = UIColor.whiteColor;
+            quotedAttachmentView = [UIView containerView];
+            [quotedAttachmentView addSubview:contentImageView];
+            quotedAttachmentView.backgroundColor = self.highlightColor;
+            quotedAttachmentView.layer.cornerRadius = self.quotedAttachmentSize * 0.5f;
+            [contentImageView autoCenterInSuperview];
+            [contentImageView
+                autoSetDimensionsToSize:CGSizeMake(self.quotedAttachmentSize * 0.5f, self.quotedAttachmentSize * 0.5f)];
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m]:
```
    contentImageView.layer.minificationFilter = kCAFilterTrilinear;
    contentImageView.layer.magnificationFilter = kCAFilterTrilinear;
```

File: Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m [hunk 3]
```diff
@@ -224,6 +243,24 @@ - (void)createContents
     }
 }
 
+- (void)didTapFailedThumbnailDownload:(UITapGestureRecognizer *)gestureRecognizer
+{
+    DDLogDebug(@"%@ in didTapFailedThumbnailDownload", self.logTag);
+
+    if (!self.quotedMessage.thumbnailDownloadFailed) {
+        OWSFail(@"%@ in %s thumbnailDownloadFailed was unexectedly false", self.logTag, __PRETTY_FUNCTION__);
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m]: unex_p_ectedly

Comment (michaelkirk-signal) [reply Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m]: fixed

File: Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.h (hunk 1 above)
Comment (michaelkirk-signal) [reply Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.h]: fixed

File: Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m (hunk 2 above)
Comment (michaelkirk-signal) [reply Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m]: fixed

**Comment by michaelkirk-signal** [comment]
In 4d539ba1c

**Pull request closed without merging**